packages/platform-browser/src/dom/shared_styles_host.ts

```diff
@@ -44,16 +44,22 @@ export class SharedStylesHost implements OnDestroy {
     }
   }
 
-  removeStyles(styles: string[]): void {
+  disableStyles(styles: string[]): void {
     for (const style of styles) {
       const usageCount = this.changeUsageCount(style, -1);
 
       if (usageCount <= 0) {
-        this.onStyleRemoved(style);
+        this.visitStyleElement(style, (node) => {
+          node.disabled = true;
+        });
```

Comment:
nit: I think it'd be great to create a couple symmetric functions to enable/disable a stylesheet. We can also use those functions to document why we disable a stylesheet instead of removing it:
```suggestion
        this.visitStyleElement(style, disableStylesheet);
```

Here is how the functions may look like:
```
/**
 * When a component that has styles is destroyed, we disable stylesheets
 * instead of removing them to avoid performance issues related to style
 * recalculation in a browser.
 */
function disableStylesheet(node: HTMLStyleElement) {
  node.disabled = true;
}

/**
 * Enables a stylesheet in a browser, see the `disableStylesheet` function
 * docs for additional info.
 */
function enableStylesheet(node: HTMLStyleElement) {
  node.disabled = false;
}
```